src/ws_transport.ts

```diff
@@ -52,29 +58,61 @@ export class WsTransport implements Transport {
     this.done = false;
     this.socketClosed = false;
     this.encrypted = false;
+    this.peeked = false;
   }
 
-  async connect(
+  connect(
     server: Server,
     options: ConnectionOptions,
   ): Promise<void> {
     const connected = false;
     const connLock = deferred<void>();
 
+    // ws client doesn't support TLS setting
+    if (options.tls) {
+      connLock.reject(new NatsError("tls", ErrorCode.INVALID_OPTION));
+      return connLock;
+    }
+
     this.options = options;
     const u = server.src;
     this.encrypted = u.indexOf("wss://") === 0;
     this.socket = new WebSocket(u);
     this.socket.binaryType = "arraybuffer";
 
     this.socket.onopen = () => {
-      this.connected = true;
-      connLock.resolve();
+      // we don't do anything here...
     };
 
     this.socket.onmessage = (me: MessageEvent) => {
       this.yields.push(new Uint8Array(me.data));
-      this.signal.resolve();
+      if (this.peeked) {
+        this.signal.resolve();
+        return;
+      }
+      const t = DataBuffer.concat(...this.yields);
+      const pm = extractProtocolMessage(t);
+      if (pm) {
+        const m = INFO.exec(pm);
+        if (!m) {
+          if (options.debug) {
+            console.error("!!!", render(t));
```

Comment:
Was it when you were debugging or planing to leave this here? If so, would use a better text

Reply:
this is only shown if you specify the debug flag which renders the protocol - normally the protocol handler prints all this stuff, but here we do. I think it is OK as this is going to be followed by an error.